packages/design-system-web/src/components/geov-yasgui/geov-yasgui.tsx

```diff
@@ -0,0 +1,144 @@
+import { importYasgui } from '../../lib/importYasgui';
+import * as MyMapPlugin from './MyMapPlugin';
+
+interface Query {
+  name?: string;
+  sparqlEndpoint: string;
+  query: string;
+}
+@Component({
+  tag: 'geov-yasgui',
+  shadow: false,
+})
+export class GeovYasgui {
+  @Element() el: HTMLElement;
+
+  @Prop() queries: Query[] = [
+    {
+      name: 'Origin of Persons',
+      sparqlEndpoint: 'https://sparql.geovistory.org/api_v1_project_591',
+      query: `PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX ontome: <https://ontome.net/ontology/>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+
+SELECT ?label ?long ?lat (count(?person) * 0.5 as ?radius) (count(?person) as ?number) ?type ?link
+    WHERE {
+
+    # Geographical Place -had presence-> Presence -was at-> Place (lat/long)
+    ?s ontome:p147i/ontome:p148 ?place.
+
+    # Geographical Place -label-> label
+    ?s rdfs:label ?label.
+
+    # Geographical Place -is origin of-> Person
+    ?s ontome:p1439i ?person.
+
+    # Geographical Place -has type-> Geographical Place Type -label-> label
+    optional{?s ontome:p1110/rdfs:label ?type}
+
+    # Extract lat and long from WKT
+  bind(replace(str(?place), '<http://www.opengis.net/def/crs/EPSG/0/4326>', "", "i") as ?rep)
+    bind(xsd:float(replace(str(?rep), "^[^0-9\\\\.-]*([-]?[0-9\\\\.]+) .*$", "$1" )) as ?long )
+    bind(xsd:float(replace( str(?rep), "^.* ([-]?[0-9\\\\.]+)[^0-9\\\\.]*$", "$1" )) as ?lat )
+
+    # Append the project query param to the URI
+  bind(concat(str(?s), "?p=591") as ?link )
+}
+GROUP BY ?label ?long ?lat ?type ?link
+`,
+    },
+    {
+      name: 'Origin of Persons2',
+      sparqlEndpoint: 'https://sparql.geovistory.org/api_v1_project_591',
+      query: `PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX ontome: <https://ontome.net/ontology/>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+
+SELECT ?label ?long ?lat (count(?person) * 0.5 as ?radius) (count(?person) as ?number) ?type ?link
+    WHERE {
+
+    # Geographical Place -had presence-> Presence -was at-> Place (lat/long)
+    ?s ontome:p147i/ontome:p148 ?place.
+
+    # Geographical Place -label-> label
+    ?s rdfs:label ?label.
+
+    # Geographical Place -is origin of-> Person
+    ?s ontome:p1439i ?person.
+
+    # Geographical Place -has type-> Geographical Place Type -label-> label
+    optional{?s ontome:p1110/rdfs:label ?type}
+
+    # Extract lat and long from WKT
+  bind(replace(str(?place), '<http://www.opengis.net/def/crs/EPSG/0/4326>', "", "i") as ?rep)
+    bind(xsd:float(replace(str(?rep), "^[^0-9\\\\.-]*([-]?[0-9\\\\.]+) .*$", "$1" )) as ?long )
+    bind(xsd:float(replace( str(?rep), "^.* ([-]?[0-9\\\\.]+)[^0-9\\\\.]*$", "$1" )) as ?lat )
+
+    # Append the project query param to the URI
+  bind(concat(str(?s), "?p=591") as ?link )
+}
+GROUP BY ?label ?long ?lat ?type ?link
+`,
+    },
+  ];
+
+  @Prop() defaultPlugin: 'map' | 'response' | 'table' = 'map';
```

Comment:
I think we need to add a new prop that tells the components which plugins to load.
availablePlugins: array of arrays: [''title', 'component-name']. So it would be for example ['map', 'geov-map-one']

I tried dynamicly importing the Plugin class after that, but this doesn't seem to work:
```
const pluginClass = testArray.map(async name => await import(`../${name}/MyMapPlugin.tsx`));
```

Can we chat about this?

Reply:
(just found a solution to this... I'm going to push my changes in a few minutes.)